pkg/graveler/graveler.go

```diff
@@ -1077,15 +1221,72 @@ func newStagingToken(repositoryID RepositoryID, branchID BranchID) StagingToken
 }
 
 func (g *KVGraveler) validateCommitParent(ctx context.Context, repositoryID RepositoryID, commit Commit) (CommitID, error) {
-	return g.db.validateCommitParent(ctx, repositoryID, commit)
+	if len(commit.Parents) > 1 {
+		return "", ErrMultipleParents
+	}
+	if len(commit.Parents) == 0 {
+		return "", nil
+	}
+
+	parentCommitID := commit.Parents[0]
+	_, err := g.RefManager.GetCommit(ctx, repositoryID, parentCommitID)
+	if err != nil {
+		return "", fmt.Errorf("get parent commit %s: %w", parentCommitID, err)
+	}
+	return parentCommitID, nil
 }
 
 func (g *KVGraveler) isCommitExist(ctx context.Context, repositoryID RepositoryID, commitID CommitID) (bool, error) {
-	return g.db.isCommitExist(ctx, repositoryID, commitID)
+	_, err := g.RefManager.GetCommit(ctx, repositoryID, commitID)
+	if err == nil {
+		// commit already exists
+		return true, nil
+	}
+	if !errors.Is(err, ErrCommitNotFound) {
+		return false, fmt.Errorf("getting commit %s: %w", commitID, err)
+	}
+	return false, nil
 }
 
 func (g *KVGraveler) AddCommitToBranchHead(ctx context.Context, repositoryID RepositoryID, branchID BranchID, commit Commit) (CommitID, error) {
```

Comment:
It seems like both you & @N-o-Z implemented that.

Reply:
yes, it is needed for the tests. we will need to solve conflicts.